include/libp2p/common/asio_buffer.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_COMMON_ASIO_BUFFER_HPP
+#define LIBP2P_COMMON_ASIO_BUFFER_HPP
+
+#include <boost/asio/buffer.hpp>
+#include <libp2p/common/span_size.hpp>
+
+namespace libp2p {
+  inline boost::asio::const_buffer asioBuffer(
+      const gsl::span<const uint8_t> &s) {
+    return {s.data(), spanSize(s)};
+  }
+
+  inline boost::asio::mutable_buffer asioBuffer(const gsl::span<uint8_t> &s) {
```

Comment:
I think is better call with copy lightweight span instead use ref (aka pointer inside). Previous reloading would be enough.

Reply:
Ok, will remove reference.

No, two overloads for `const_buffer` and `mutable_buffer` are required